CollAction/Services/Donation/DonationService.cs

```diff
@@ -0,0 +1,165 @@
+﻿using CollAction.Data;
+using CollAction.Models;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+using Stripe;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CollAction.Services.Donation
+{
+    public class DonationService : IDonationService
+    {
+        private readonly CustomerService _customerService;
+        private readonly SourceService _sourceService;
+        private readonly ApplicationDbContext _context;
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly RequestOptions _requestOptions;
+        private readonly SiteOptions _siteOptions;
+
+        public DonationService(IOptions<RequestOptions> requestOptions, IOptions<SiteOptions> siteOptions, UserManager<ApplicationUser> userManager, ApplicationDbContext context)
+        {
+            _requestOptions = requestOptions.Value;
+            _siteOptions = siteOptions.Value;
+            _customerService = new CustomerService(_requestOptions.ApiKey);
+            _sourceService = new SourceService(_requestOptions.ApiKey);
+            _context = context;
+            _userManager = userManager;
+        }
+
+        public async Task<string> InitializeCreditCardCheckout(string currency, int amount, string name, string email)
+        {
+            ValidateDetails(amount, name, email);
+
+            ApplicationUser user = await _userManager.FindByEmailAsync(email);
+            Customer customer = await GetOrCreateCustomer(name, email);
+
+            using (var client = new HttpClient())
+            {
+                var requestContent = new Dictionary<string, string>()
+                {
+                    { "success_url", $"{_siteOptions.PublicAddress}/Donation/ThankYou" },
+                    { "cancel_url", $"{_siteOptions.PublicAddress}/Donation/Donate" },
```

Comment:
That's interesting... it there no cancel_url for ideal payments? (see my earlier comment)

Reply:
This is a beta api, so a bit new.